.github/workflows/deploy-webui.yml

```diff
@@ -0,0 +1,51 @@
+name: Deploy webui to Github Pages
+
+on:
+  # Runs on pushes targeting the webui branch
+  push:
+    branches: ['webui']
+
+  # Allows to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets the GITHUB_TOKEN permissions to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: 'pages'
+  cancel-in-progress: true
+
+jobs:
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
+          cache-dependency-path: './webui/package-lock.json'
+      - name: Install dependencies
+        run: npm ci
+        working-directory: ./webui
+      - name: Build
+        run: npm run build
+        working-directory: ./webui
+      - name: Setup Pages
+        uses: actions/configure-pages@v4
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: './webui/dist'
+      - name: Deploy to GitHub Pages
```

Comment:
we'll have to figure out how to merge multiple apps, but i don't anticipate this to be hard, just maybe tedious. no action right now.